src/complex/Parameters/FileSystemPathParameter.hpp

```diff
@@ -25,10 +26,12 @@ class COMPLEX_EXPORT FileSystemPathParameter : public ValueParameter
     OutputDir = 3
   };
 
+  using ExtensionsType = std::unordered_set<std::string>;
+
   using ValueType = std::filesystem::path;
 
   FileSystemPathParameter() = delete;
-  FileSystemPathParameter(const std::string& name, const std::string& humanName, const std::string& helpText, const ValueType& defaultValue, PathType pathType);
+  FileSystemPathParameter(const std::string& name, const std::string& humanName, const std::string& helpText, const ValueType& defaultValue, PathType pathType, ExtensionsType extensionsType={});
```

Comment:
Consider making the new parameter a const reference to avoid copying data

Reply:
Ah good catch thanks.